airbyte-workers/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java

```diff
@@ -44,6 +45,13 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(KubeProcessFactory.class);
 
+  public static final String JOB_TYPE = "job_type";
+  public static final String SYNC_JOB = "sync";
+  public static final String SPEC_JOB = "spec";
+  public static final String CHECK_JOB = "check";
+  public static final String DISCOVER_JOB = "discover";
+  public static final String NORMALIZATION_JOB = "normalize";
```

Comment:
This would serve to track the resources consumed by the kube pods and classify by job type, right?

Shouldn't there be an additional one for custom transformation too? (separate from `normalize`?

In addition, in case of normalization and transformations, the pods are used to trigger dbt cli (which then makes calls to the data warehouse engine) but, these processes don't necessarily require much CPU/memory resources themselves because all the transformation work is actually offloaded and performed by the destination engine (not the Kube pod itself). 
As a result, we would also need to track some credit differently there to have a good view of how much the job cost?

Reply:
Yes this is so we can track how much it costs in our cloud.

I didn't add any for the custom dbt runner (the map if empty) since this doesn't work for Kube today. I can do so.

Yeap. We aren't including normalisation in the initial pricing since we haven't optimised it yet - that's part of the reason for the separate labelling. As for how to track the cost - I'm not sure since we don't have a good view in to the end warehouse where the DBT commands are being executed in. I also think this is outside the scope of our tracking (this is focused on Cloud cost tracking)

Although agree we should try to get an estimate.

Reply:
> Although agree we should try to get an estimate.

There should probably be a dedicated issue for this (outside of this PR/effort)...

Some data on what the dbt cost could be estimated from here:
> adapter_response: Dictionary of metadata returned from the database, which varies by adapter. E.g. success code, number of rows_affected, total bytes_processed, etc. Not populated by tests, as of v0.19.0; we plan to fix in a future release
https://docs.getdbt.com/reference/artifacts/run-results-json#top-level-keys

Reply:
created issue: https://github.com/airbytehq/airbyte/issues/6372